ludwig/hyperopt/execution.py

```diff
@@ -818,7 +843,36 @@ def get_build_hyperopt_executor(executor_type):
 executor_registry = {"ray": RayTuneExecutor}
 
 
-def set_values(model_dict, name, parameters_dict):
+def set_values(
```

Comment:
While this is technically covered by the larger end to end integration test, it might be good to add unit tests for this function explicitly since what this function is doing is already rather non-trivial.

Reply:
Agreed! I've decided to split this into two functions - one that just sets value for normal parameters from the sampler, and another that sets values for shared parameters. This will also let me get rid of the deeply nested conditionals. Will update :) 

Will try and add unit tests for my new simplified functions